packages/internal/src/build/babel/api.ts

```diff
@@ -28,7 +28,14 @@ export const getApiSideBabelPresets = (
   { presetEnv } = { presetEnv: false }
 ) => {
   return [
-    '@babel/preset-typescript',
+    [
+      '@babel/preset-typescript',
+      {
+        isTSX: true,
+        allExtensions: true,
+      },
+      'rwjs-babel-preset-typescript',
+    ],
```

Comment:
This is the change I'm least knowledgeable about and so the most unsure about. I did try various configurations but this was the key to getting JSX working on the API side of JS projects.